src/connectors/configure/connectConfigure.ts

```diff
@@ -88,11 +88,11 @@ const connectConfigure: ConfigureConnector = function connectConfigure(
           new algoliasearchHelper.SearchParameters(searchParameters)
         );
 
-        // Trigger a search with the resolved search parameters
-        helper.setState(nextSearchParameters).search();
-
         // Update original `widgetParams.searchParameters` to the new refined one
         widgetParams.searchParameters = searchParameters;
+
+        // Trigger a search with the resolved search parameters
+        helper.setState(nextSearchParameters).search();
```

Comment:
When `search` happens, it will trigger `onStateChange`.

![image](https://user-images.githubusercontent.com/499898/97614670-82270800-1a1a-11eb-8889-af1b09b27617.png)

Before this change, the `widgetParams` was outdated at this point, now it's updated first, then the search is triggered.